projects/observability/src/shared/services/filter-builder/graphql-filter-builder.service.ts

```diff
@@ -1,13 +1,36 @@
+import { FieldFilter, FilterValue } from './../../../../../components/src/filtering/filter/filter';
+import { TableFilter } from './../../../../../components/src/table/table-api';
 import { Injectable } from '@angular/core';
 import { assertUnreachable } from '@hypertrace/common';
-import { FilterOperator, TableFilter } from '@hypertrace/components';
+import { FilterOperator } from '@hypertrace/components';
 import { GraphQlArgumentValue } from '@hypertrace/graphql-client';
 import { GraphQlFieldFilter } from '../../graphql/model/schema/filter/field/graphql-field-filter';
 import { GraphQlFilter, GraphQlOperatorType } from '../../graphql/model/schema/filter/graphql-filter';
 
 @Injectable({ providedIn: 'root' })
 export class GraphQlFilterBuilderService {
-  public buildGraphQlFilters(filters: TableFilter[]): GraphQlFilter[] {
+  public buildFiltersFromGraphQlFieldFilters(filters: GraphQlFieldFilter[]): FieldFilter[] {
```

Comment:
Why do we want to convert back from GQL filters to field filters? If we're doing refactoring, we should change any components using GQL filters to use the UI construct instead of supporting going bidrectionally.

Reply:
Yeah, so that would be a big change. If we want to do it in iterations then temporarily this method will be used in the basic filter bar for now. Slowly, we will migrate the filters to the UI construct. 

We would also need to revisit the relationship between `IncompleteFilter`, `FieldFilter`, and `Filter`?

Reply:
Things I am re-using:
1. Filter Objects and various filter builders 
2. Url string to Filter parsing logic
3. Basic filter bar value auto-completion

Things I am changing:
1. Moving UI side code to depend on `Filter` objects instead of GraphQlFilters. We will use a method that would have the conversion logic. This migration will have in phases and we can track it with stories.
2. Fix the dependency of Filter, FieldFilter, and IncompleteFilter
2. Using new design for basic view


@jake-bassett Migrating to UI construct (point 1) has a much bigger blast radius. We should do it in stages. Let's wait for other higher-priority tasks to get completed first.

Reply:
Okay, got it. Yeah, the items you are re-using are in desperate need of rework but I know that's not at all trivial. I'm good with doing this in phases. As much as I know that code is hard to work with, and making things a bit of a challenge, I don't think we can wait on these filter changes much longer.

I vote for not making this PR any larger that what it already is and revisiting replacing the rest of the filtering at a later date. Would appreciate if @aaron-steinfeld can chime in on any more quick wins we can do here in the meantime though.

Reply:
okay. approve this PR  :)